### PR TITLE
[Filebeat] change multiline configuration in awss3 input to parsers

### LINKED ISSUE #25249: [Filebeat] Support multiline options in aws s3 input
**Describe the enhancement:**

When reading log files from S3 users should be able to specify the same `multiline` options that are available with the `log` input.

**Describe a specific use case for the enhancement or feature:**

Reading XML based Windows event logs from S3 that are newline delimited, but the XML itself contains strings with newlines. So in order to get one full XML object we need the multiline reader options.

Ideally config like this would work:

```
- type: aws-s3
  queue_url: https://sqs.us-east-1.amazonaws.com/foo/queue
  credential_profile_name: beats
  multiline.pattern: ^\<Event\>
  multiline.negate: true
  multiline.match: after
```

### PR DESCRIPTION
**Pull request by leehinman**
## What does this PR do?
Changes configuration of multiline in awss3 input to parsers to match multiline configuration in `filestream` input.

## Why is it important?

This is the new paradigm for configuring multiline and we want consistency between inputs.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
~~- [ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

- Setup S3 & SQS in AWS
- Set QUEUE_URL shell variable
- Set AWS_PROFILE_NAME shell variable
- Set S3_BUCKET_NAME shell variable
- Set S3_BUCKET_REGION shell variable
- Run `go test --tags=integration,aws`

## Related issues

- Closes #25249

### CODE REVIEW

File: x-pack/filebeat/input/awss3/parser.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	io.Closer
+	Next() (reader.Message, error)
+}
+
+type parserConfig struct {
+	maxBytes       cfgtype.ByteSize
+	lineTerminator readfile.LineTerminator
+}
+
+func newParsers(in reader.Reader, pCfg parserConfig, c []common.ConfigNamespace) (parser, error) {
```
Comment (urso) [inline x-pack/filebeat/input/awss3/parser.go:35]: @kvch this looks like we have to duplicate parser setup from the filestream input here. This is somewhat unfortunate, especially if we want to add similar functionality to other inputs. How about providing a common package for parser setup?

Inputs that want to make use of parsers should be able to do it more or less like this:


```

type Config struct {
  ...

  Parser inpparser.Settings
  ...
}

...

parser, err := config.Parser.Create(reader)
...

```

**Comment by leehinman** [comment]
needs to be rebased onto #26541 when that is merged

**Review by andrewkroh — APPROVED** [review]
LGTM

File: x-pack/filebeat/docs/inputs/input-aws-s3.asciidoc [hunk 2]
```diff
@@ -166,15 +166,43 @@ The maximum number of messages to return. Amazon SQS never returns more messages
...
+- type: {type}
+  ...
+  parsers:
+    - multiline:
+        pattern: "^<Event"
+        negate:  true
+        match:   after
+----
+
+See the available parser settings in detail below.
```
Comment (jsoriano) [inline x-pack/filebeat/docs/inputs/input-aws-s3.asciidoc:196]: @kvch would it be possible/make sense to have a shared asciidoc that can be included in all the inputs with `parsers`?

**Pull request merged**